src/parser/jobs/nin/Ninki.js

```diff
@@ -0,0 +1,102 @@
+import React, {Fragment} from 'react'
+//import {Icon, Message} from 'semantic-ui-react'
+
+//import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+//import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+// Constants
+const MAX_NINKI = 100
+const NINKI_SPENDERS = [
+	ACTIONS.HELLFROG_MEDIUM.id,
+	ACTIONS.BHAVACAKRA.id,
+	ACTIONS.TEN_CHI_JIN.id,
+]
+
+export default class Ninki extends Module {
+	static handle = 'ninki'
+	static dependencies = [
+		'suggestions',
+	]
+	
+	_ninki = 0
+	_wastedNinki = 0
+	_wasteBySource = {
+		mug: 0,
+		auto: 0,
+	}
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		//this.addHook('death', {to: 'player'}, this._onDeath)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onCast(event) {
+		const abilityId = event.ability.guid
+		
+		if (abilityId === ACTIONS.MUG.id) {
+			this._wasteBySource.mug += this._addNinki(30)
+		}
+
+		if (abilityId === ACTIONS.ATTACK.id) {
+			this._wasteBySource.auto += this._addNinki(6)
+		}
+
+		if (NINKI_SPENDERS.includes(abilityId)) {
+			this._ninki -= 80
+		}
+	}
+
+	_addNinki(amt) {
+		// Helper for adding Ninki to the running tally and calculating waste. Returns the amount wasted.
+		this._ninki += amt 
+		if (this._ninki > MAX_NINKI) {
+			let waste = this._ninki - MAX_NINKI
+			this._wastedNinki += waste
+			this._ninki = MAX_NINKI
+			return waste 
+		}
+
+		return 0 
+	}
+
+/* Not used for now, but perhaps in the future
+	_onDeath() {
+		// YOU DONE FUCKED UP NOW
+		this._wastedNinki += this._ninki
+		this._wasteBySource.death += this._ninki
+		this._ninki = 0
+	}
+*/
+
+	_onComplete() {
+		if (this._wastedNinki >= 20) {
```

Comment:
Not going to pretend I know a huge amount about NIN, but this _seems_ a bit high?

Probably worth reducing the minimum for the suggestion, and adding a dynamic `severity` to show how bad it is/isn't.

Reply:
It was admittedly a somewhat arbitrary decision. The main reason to not make it super low is that there are some occasions when holding into overcap is the desired thing to do (mainly when you use the Bhava opener and then hold your second spender as TCJ -> Suiton -> TA). The theorycrafters may have better final numbers to use, but I'm thinking for now I can change it to 12 as the threshold for MINOR and then bump it to, let's say 24 for MEDIUM.

Reply:
Sounds good, let's go with that for now then.

Reply:
:+1: